### PR TITLE
Added first two conventions on the guidelines

### PR DESCRIPTION
**Pull request by agerace**
Before merging this PR, go over this checklist:
- [ ] The issue was approved by a considerable number of votes
- [ ] The rule text respects the structure:
    - [ ] Title
    - [ ] Short Description
    - [ ] Examples
    - [ ] Reasoning

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+      "commentCount": 0,
+      "createdAt": "2017-03-07T18:57:44.622Z",
+      "id": "string",
+      "likeCount": 0,
+      "mediaType": "image",
+      "owner": "string"
+    }
+
+##### Reasoning
+Even though in some programming languages is common to name variables or object instances with camelCase names, we consider the snake_case is the most readable and understandable way to named the key fields on a JSON response
```
Comment (volbap) [inline README.md]: (...) we consider the snake_case is the most readable and understandable way to **name** the key fields on a JSON response

File: README.md [hunk 2]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
 
 ## Conventions & Rules
 
-#### Maintain existing style
```
Comment (volbap) [inline README.md]: @agerace We do not want to delete this one! :P

**Review by volbap — CHANGES_REQUESTED** [review]
Great! Good start. Just a couple of little fixes are needed.

File: README.md (hunk 2 above)
Comment (elbrujohalcon) [reply README.md]: I agree with @volbap

File: README.md [hunk 3]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
-> When editing a system written by someone else, stick to the style in which it was designed. If a project has an overall style, stick to that when adding, removing or updating API endpoints as well.
+#### Use snake_case with JSON
+> When the endpoint returns a JSON, the keys should always be in snake_case, unless you have a super edge case.
```
Comment (elbrujohalcon) [inline README.md]: No unles'es in the guidelines. Exceptions do exist, but the guidelines define the good scenario only.

Comment (elbrujohalcon) [reply README.md]: That's why they are *guidelines* and not *rules*

File: README.md (hunk 1 above)
Comment (elbrujohalcon) [reply README.md]: on a JSON response -> on a JSON object

File: README.md [hunk 4]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+
 
-###### Ugly
-- **POST** `/set_table`
-- **GET** `/tables/x`
-- **POST** `/api/v2.3/set_chair` ⃪ _added by you_
+
+
+#### Don't use response envelopes by default
+> When the endpoint returns, for example, an array, it should be returned as a plain array, not a dictionary with the array as a value.
```
Comment (elbrujohalcon) [inline README.md]: dictionary -> object

dictionaries in JSON are just called _objects_

File: README.md (hunk 2 above)
Comment (agerace) [reply README.md]: oh, upsi... my bad? jajaja 😅

File: README.md [hunk 5]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+          "id": "string",
+          "media_type": "comment",
+          "owner": "string"
+        }
+      ]
+    }
 
 ##### Reasoning
-It's better to maintain a system that just looks ugly to you than to have a system that looks half ugly to you, half ugly to somebody else.
+If the client expects an array it could ( and would ) be confusing to receive a dictionary with one key containing the array as a value.
```
Comment (elbrujohalcon) [inline README.md]: This is not the proper reasoning. The reasoning behind this rule is _semantics_.
If you ask `GET /entities`, the natural meaning of that is _**GET** me all the **entities**_ not _**GET** me an object that contains all the **entities** in one of its properties_. There are scenarios when that is desired, but we should always default to the implementation that's most intuitive.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sheldonci) [inline README.md]: I found 3 misspelled words in this sentence: 
- JSON
	- *Suggested Replacements:* son, jason, jeon
- snake_case
- camelCase

File: README.md [hunk 6]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+        {
+          "like_count": 0,
+          "id": "string",
+          "media_type": "comment",
+          "owner": "string"
+        }
+      ]
+    }
 
 ##### Reasoning
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md [hunk 7]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
 
 ## Conventions & Rules
```
Comment (sheldonci) [inline README.md]: I found 2 misspelled words in this sentence: 
- &
	- *Suggested Replacements:* , k, p, u, e, z, j, &c, o, t, d, y, i, n, s, c, x, h, m, r, b, w, g, l, q, a, v, f
- ##

File: README.md [hunk 8]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
 
-#### Maintain existing style
-> When editing a system written by someone else, stick to the style in which it was designed. If a project has an overall style, stick to that when adding, removing or updating API endpoints as well.
+#### Use snake_case with JSON
```
Comment (sheldonci) [inline README.md]: I found 3 misspelled words in this sentence: 
- JSON
	- *Suggested Replacements:* son, jason, jeon
- snake_case
- ####

File: README.md (hunk 3 above)
Comment (sheldonci) [inline README.md]: I found 3 misspelled words in this sentence: 
- snake_case
- JSON
	- *Suggested Replacements:* son, jason, jeon
- >
	- *Suggested Replacements:* , k, p, u, e, z, j, o, t, d, y, i, n, s, c, x, h, m, r, b, w, g, l, q, a, v, f

Comment (sheldonci) [inline README.md]: I found 3 misspelled words in this sentence: 
- snake_case
- JSON
	- *Suggested Replacements:* son, jason, jeon
- >
	- *Suggested Replacements:* , k, p, u, e, z, j, o, t, d, y, i, n, s, c, x, h, m, r, b, w, g, l, q, a, v, f

File: README.md [hunk 9]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+      "caption": "string",
+      "commentCount": 0,
+      "createdAt": "2017-03-07T18:57:44.622Z",
+      "id": "string",
+      "likeCount": 0,
+      "mediaType": "image",
+      "owner": "string"
+    }
+
+##### Reasoning
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md [hunk 10]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
-- **POST** `/set_table`
-- **GET** `/tables/x`
-- **POST** `/api/v2.3/set_chair` ⃪ _added by you_
+
+
+#### Don't use response envelopes by default
+> When the endpoint returns, for example, an array, it should be returned as a plain array, not a dictionary with the array as a value.
+
+
+##### Examples
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md [hunk 11]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+          "owner": "string"
+        }
+      ]
+    }
 
 ##### Reasoning
-It's better to maintain a system that just looks ugly to you than to have a system that looks half ugly to you, half ugly to somebody else.
+If the client expects an array it could ( and would ) be confusing to receive a dictionary with one key containing the array as a value. 
 
 ***
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ***

File: README.md [hunk 12]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
 ## Conventions & Rules
 
-#### Maintain existing style
-> When editing a system written by someone else, stick to the style in which it was designed. If a project has an overall style, stick to that when adding, removing or updating API endpoints as well.
+#### Use snake_case with JSON
+> When the endpoint returns a JSON, the keys should always be in snake_case, unless you have a super edge case.
+
 
 ##### Examples
 ###### Good
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md (hunk 11 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ***

File: README.md (hunk 10 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md [hunk 13]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
-- **GET** `/tables/x`
-- **POST** `/api/v2.3/set_chair` ⃪ _added by you_
+
+
+#### Don't use response envelopes by default
+> When the endpoint returns, for example, an array, it should be returned as a plain array, not a dictionary with the array as a value.
+
+
+##### Examples
+###### Good
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md (hunk 1 above)
Comment (sheldonci) [inline README.md]: I found 3 misspelled words in this sentence: 
- JSON
	- *Suggested Replacements:* son, jason, jeon
- snake_case
- camelCase

File: README.md [hunk 14]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+      },
+      {
+        "like_count": 0,
+        "id": "string",
+        "media_type": "comment",
+        "owner": "string"
+      }
+    ]
+
+###### Bad
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md (hunk 9 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md [hunk 15]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+Even though in some programming languages is common to name variables or object instances with camelCase names, we consider the snake_case is the most readable and understandable way to named the key fields on a JSON response
+
 
-###### Ugly
-- **POST** `/set_table`
-- **GET** `/tables/x`
-- **POST** `/api/v2.3/set_chair` ⃪ _added by you_
+
+
+#### Don't use response envelopes by default
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ####

File: README.md (hunk 6 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md (hunk 12 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md (hunk 13 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md (hunk 7 above)
Comment (sheldonci) [inline README.md]: I found 2 misspelled words in this sentence: 
- &
	- *Suggested Replacements:* , k, p, u, e, z, j, &c, o, t, d, y, i, n, s, c, x, h, m, r, b, w, g, l, q, a, v, f
- ##

File: README.md [hunk 16]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
 
 ## Conventions & Rules
 
-#### Maintain existing style
-> When editing a system written by someone else, stick to the style in which it was designed. If a project has an overall style, stick to that when adding, removing or updating API endpoints as well.
+#### Use snake_case with JSON
+> When the endpoint returns a JSON, the keys should always be in snake_case, unless you have a super edge case.
+
 
 ##### Examples
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md [hunk 17]
```diff
@@ -15,30 +15,97 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+      "caption": "string",
+      "comment_count": 0,
+      "created_at": "2017-03-07T18:57:44.622Z",
+      "id": "string",
+      "like_count": 0,
+      "media_type": "image",
+      "owner": "string"
+    }
 
 ###### Bad
```
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md (hunk 8 above)
Comment (sheldonci) [inline README.md]: I found 3 misspelled words in this sentence: 
- JSON
	- *Suggested Replacements:* son, jason, jeon
- snake_case
- ####

File: README.md (hunk 17 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md (hunk 16 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- #####

File: README.md (hunk 4 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- >
	- *Suggested Replacements:* , k, p, u, e, z, j, o, t, d, y, i, n, s, c, x, h, m, r, b, w, g, l, q, a, v, f

File: README.md (hunk 15 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ####

File: README.md (hunk 14 above)
Comment (sheldonci) [inline README.md]: I found 1 misspelled word in this sentence: 
- ######

File: README.md [hunk 18]
```diff
@@ -102,7 +102,7 @@ Even though in some programming languages is common to name variables or object
     }
 
 ##### Reasoning
-If the client expects an array it could ( and would ) be confusing to receive a dictionary with one key containing the array as a value. 
+If you're calling a **GET** on `/entities` , you're asking for entities, not an object with the entities inside one of its properties. Think about it as a christmas gift. If you ask santa for all the power rangers action figures ( and i'm talking about the 90's ones ) and you only see one box under the tree, you can't know if you're getting just one or all of them wrapped in one gift. _Also,  ALL the power rangers? Couldn't you just ask for ONE toy?_
```
Comment (elbrujohalcon) [inline README.md]: HA HA! Very funny… not sure if that's the spirit we want to showcase in this repos. @volbap what do you think?

File: README.md [hunk 19]
```diff
@@ -11,6 +11,129 @@ For **questions** or **general comments** regarding the use of this repository,
...
+      ]
+    }
+
+##### Reasoning
+If you're calling a **GET** on `/entities` , you're asking for entities, not an object with the entities inside one of its properties. Think about it as a christmas gift. If you ask santa for all the power rangers action figures ( and i'm talking about the 90's ones ) and you only see one box under the tree, you can't know if you're getting just one or all of them wrapped in one gift. _Also,  ALL the power rangers? Couldn't you just ask for ONE toy?_
+
+---
+
+#### Use all HTTP verbs ( Not everything is a GET )
+> Use all the HTTP verbs. If you have to Create something use POST. If you want to Eead Something. if you want to Update something use PATCH or PUT. If you want to Delete something use DELETE. Always remember C.R.U.D. ( Create, Read, Update, Delete) when setting the verb for your resource
```
Comment (elbrujohalcon) [inline README.md]: Eead ?

File: README.md [hunk 20]
```diff
@@ -11,6 +11,129 @@ For **questions** or **general comments** regarding the use of this repository,
...
+Creating a comment on a media item: 
+- **GET** `/media/:media_id/create_comment?text=comment%20text`
+  Updating ( editing ) the comment content on a media item:
+- **GET**  `/media/:media_id/update_comment?text=comment%20new%20text`
+  Deleting a comment on a media item:
+- **GET** `/media/:media_id/delete_comment?id=comment1`
+
+
+##### Reasoning
+Using the right HTTP Verb on each resource will make clearer what the resource is made for and what we should expect it to do, receive and return. Using always GET is like using your spoon to eat your soup and cut your steak. Just don't.
```
Comment (elbrujohalcon) [inline README.md]: Add something about how you can _read_ or _write_ your intentions, e.g.:

* **PUT** /entities/1 : it puts the entity #1 in the server
* **GET** /update_entity?id=1&title=t : it gets all the update_entities with id = 1 and title = t … WAT?

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: README.md (hunk 19 above)
Comment (agerace) [reply README.md]: Read jajaja

File: README.md (hunk 18 above)
Comment (agerace) [reply README.md]: jajaja yeah,  I was thinking the same when I write but I jumped into the pool :P

Comment (volbap) [reply README.md]: Hahahaha 🤣  Although it made me laugh out loud, we should **not** include these kind of comments in our guidelines. Sorry, @agerace , but I encourage removing it.

Comment (volbap) [reply README.md]: Maybe you could keep them in mind for whenever you write a related blogpost ;)

File: README.md [hunk 21]
```diff
@@ -11,6 +11,129 @@ For **questions** or **general comments** regarding the use of this repository,
...
+        }
+      ]
+    }
+
+##### Reasoning
+If you're calling a **GET** on `/entities` , you're asking for entities, not an object with the entities inside one of its properties. Think about it as a christmas gift. If you ask santa for all the power rangers action figures ( and i'm talking about the 90's ones ) and you only see one box under the tree, you can't know if you're getting just one or all of them wrapped in one gift. _Also,  ALL the power rangers? Couldn't you just ask for ONE toy?_
+
+---
+
+#### Use all HTTP verbs ( Not everything is a GET )
```
Comment (volbap) [inline README.md]: Are these spaces after and before the parenthesis on purpose? I would suggest removing them, and writing them the normal way: `Use all HTTP verbs (Not everything is a GET)`
Thoughts @elbrujohalcon ?

**Review by volbap — DISMISSED** [review]

File: README.md (hunk 19 above)
Comment (elbrujohalcon) [reply README.md]: Then . and , are wrong… I bet you wanted to say:
> Use all the HTTP verbs. If you have to Create something use POST. If you want to Read something use GET. If you want to Update something use PATCH or PUT. If you want to Delete something use DELETE. Always remember C.R.U.D. (Create, Read, Update, Delete) when setting the verb for your resource.

File: README.md (hunk 18 above)
Comment (agerace) [inline README.md]: @volbap  ?

Comment (agerace) [reply README.md]: roger that

File: README.md (hunk 19 above)
Comment (agerace) [reply README.md]: ups, yes, I forgot the "use GET" there.

File: README.md [hunk 22]
```diff
@@ -102,42 +102,52 @@ Even though in some programming languages is common to name variables or object
     }
 
 ##### Reasoning
-If you're calling a **GET** on `/entities` , you're asking for entities, not an object with the entities inside one of its properties. Think about it as a christmas gift. If you ask santa for all the power rangers action figures ( and i'm talking about the 90's ones ) and you only see one box under the tree, you can't know if you're getting just one or all of them wrapped in one gift. _Also,  ALL the power rangers? Couldn't you just ask for ONE toy?_
+If you're calling a **GET** on `/entities` , you're asking for entities, not an object with the entities inside one of its properties. So don't envelope things that should be being retrieved directly on the object.
```
Comment (elbrujohalcon) [inline README.md]: ...you are asking for a list of entities, not... ....things that should be retrieved directly.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: README.md [hunk 23]
```diff
@@ -11,10 +11,143 @@ For **questions** or **general comments** regarding the use of this repository,
...
+---
+
+#### Don't use response envelopes by default
+> When the endpoint returns, for example, an array, it should be returned as a plain array, not a dictionary with the array as a value.
+
+
+##### Examples
+###### Good
+
+    [
```
Comment (elbrujohalcon) [inline README.md]: You should add backticks around the code here, something like
```js
this
```

File: README.md [hunk 24]
```diff
@@ -11,10 +11,143 @@ For **questions** or **general comments** regarding the use of this repository,
...
+        "like_count": 0,
+        "id": "string",
+        "media_type": "comment",
+        "owner": "string"
+      }
+    ]
+
+###### Bad
+
+    { "items" : [
```
Comment (elbrujohalcon) [inline README.md]: You should add backticks around the code here, something like
```js
this
```

File: README.md [hunk 25]
```diff
@@ -11,10 +11,143 @@ For **questions** or **general comments** regarding the use of this repository,
 
 And you can check all of our open-source projects at [inaka.github.io](http://inaka.github.io)
 
----
+
 
 ## Conventions & Rules
 
+#### Use snake_case with JSON
+> When the endpoint returns a JSON, the keys should always be in snake_case.
```
Comment (elbrujohalcon) [inline README.md]: This should also apply to JSONS in the request, not just the response

File: README.md [hunk 26]
```diff
@@ -11,10 +11,143 @@ For **questions** or **general comments** regarding the use of this repository,
...
+
+
+##### Examples
+###### Good
+
+      {
```
Comment (elbrujohalcon) [inline README.md]: You should add backticks around the code here, something like
```js
this
```

File: README.md [hunk 27]
```diff
@@ -11,10 +11,143 @@ For **questions** or **general comments** regarding the use of this repository,
...
+      "created_at": "2017-03-07T18:57:44.622Z",
+      "id": "string",
+      "like_count": 0,
+      "media_type": "image",
+      "owner": "string"
+    }
+
+###### Bad
+
+      {
```
Comment (elbrujohalcon) [inline README.md]: You should add backticks around the code here, something like
```js
this
```

File: README.md [hunk 28]
```diff
@@ -11,10 +11,143 @@ For **questions** or **general comments** regarding the use of this repository,
...
+          "like_count": 0,
+          "id": "string",
+          "media_type": "comment",
+          "owner": "string"
+        }
+      ]
+    }
+
+##### Reasoning
+If you're calling a **GET** on `/entities` , you're asking for a list of entities, not an object with the entities inside one of its properties. So don't envelope things that should be retrieved directly.
```
Comment (elbrujohalcon) [inline README.md]: extra space before ,

File: README.md [hunk 29]
```diff
@@ -11,10 +11,143 @@ For **questions** or **general comments** regarding the use of this repository,
...
+
+- **GET** `/media/:media_id/delete_comment?id=comment1`
+
+
+##### Reasoning
+Using the right HTTP Verb on each resource will make clearer what the resource is made for and what we should expect it to do, receive and return. 
+
+---
+
 #### Maintain existing style
```
Comment (elbrujohalcon) [inline README.md]: I would like this to be the first guideline in the list, if possible

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]
Last round of comments, I promise

File: README.md (hunk 29 above)
Comment (agerace) [reply README.md]: everything it's possible if you believe hard enough @elbrujohalcon !

File: README.md [hunk 30]
```diff
@@ -32,13 +34,155 @@ And you can check all of our open-source projects at [inaka.github.io](http://in
...
+#### Use snake_case with JSON
+
+> When the endpoint returns or receive a JSON, the keys should always be in snake_case.
+
+
+##### Examples
+###### Good
+```
+{
+	"caption": "string",
```
Comment (elbrujohalcon) [inline README.md:59]: indentation is broken here

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

**Review by elbrujohalcon — APPROVED** [review]

**Pull request merged**